src/main/java/com/sparta/zlzonedelivery/product/service/ProductService.java

```diff
@@ -0,0 +1,134 @@
+                .name(product.getName())
+                .productCategory(product.getProductCategory().getProductCategoryName())
+                .build();
+    }
+
+    public Page<ProductReadResponseDto> getProductAll(UUID storeId, Pageable pageable) {
+
+        Store store = storeRepository.findByIdAndIsPublicIsTrue(storeId)
+                .orElseThrow(() -> new CustomException(ErrorCode.STORE_NOT_FOUND));
+
+        return productRepository.findAllByStoreAndIsPublicIsTrue(store, pageable)
+                .map(product -> ProductReadResponseDto.builder()
+                        .productId(product.getId())
+                        .name(product.getName())
+                        .description(product.getDescription())
+                        .price(product.getPrice())
+                        .productCategory(product.getProductCategory().getProductCategoryName())
+                        .build());
+    }
+
+    @Transactional
+    public void updateProduct(UUID storeId, UUID productId, ProductUpdateRequestDto requestDto) {
+
+        Product product = productRepository.findByIdAndIsPublicIsTrue(productId)
+                .orElseThrow(() -> new CustomException(ErrorCode.MENU_NOT_FOUND));
+
+        if (!storeId.equals(product.getStore().getId())) {
+            throw new CustomException(ErrorCode.INVALID_STORE);
+        }
+
+        product.updateProduct(requestDto.name(), requestDto.description(),
+                requestDto.price(), requestDto.isShown());
+
+    }
+
+    @Transactional
+    public void deleteProduct(UUID storeId, UUID productId) {
+
+        Product product = productRepository.findByIdAndIsPublicIsTrue(productId)
+                .orElseThrow(() -> new CustomException(ErrorCode.MENU_NOT_FOUND));
+
+        if (!storeId.equals(product.getStore().getId())) {
+            throw new CustomException(ErrorCode.INVALID_STORE);
+        }
+
+        productRepository.deleteById(product.getId());
+    }
+
+    public Page<ProductReadResponseDto> searchByCategory(UUID storeId, String categoryName, Pageable pageable) {
+
+        Store store = storeRepository.findByIdAndIsPublicIsTrue(storeId)
+                .orElseThrow(() -> new CustomException(ErrorCode.STORE_NOT_FOUND));
+
+        ProductCategory productCategory = productCategoryRepository.findByProductCategoryNameAndIsPublicIsTrue(categoryName)
+                .orElseThrow(() -> new CustomException(ErrorCode.PRODUCT_CATEGORY_NOT_FOUND));
+
+        return productRepository.searchByProductCategoryAndIsPublicIsTrueOrderByPrice(productCategory, pageable)
+                .map(product -> ProductReadResponseDto.builder()
+                        .productId(product.getId())
+                        .name(product.getName())
+                        .description(product.getDescription())
+                        .price(product.getPrice())
+                        .productCategory(product.getProductCategory().getProductCategoryName())
+                        .build());
```

Comment:
dto 클래스의 생성자 파라미터로 엔티티 클래스를 넣어 dto 클래스 내부에서 값을 할당해주는것은 어떻게 생각하시나요?
예를 들면 이런 느낌?
```suggestion
                .map(product -> ProductReadResponseDto::fromEntity);
```

Reply:
아 알려주셔서 감사합니다 ㅋㅋㅋ 바로 수정했습니다!